_format/1.1/index.md

```diff
@@ -27,35 +38,179 @@ document are to be interpreted as described in
+other than `ext` and `profile`.
+
+Extensions and profiles are each uniquely identified by a
+[URI](https://tools.ietf.org/html/rfc3986). Visiting an extension's or profile's
+URI **SHOULD** return documentation that describes its usage.
+
+The `ext` parameter is used to support [extensions] and the `profile` parameter
+is used to support [profiles]. The values of the `ext` and `profile` parameters
+**MUST** equal a space-separated (U+0020 SPACE, " ") list of extension or
+profile URIs, respectively.
+
+> Note: When serializing the `ext` or `profile` media type parameters, the HTTP
+> specification requires that parameter values be surrounded by quotation marks
+> (U+0022 QUOTATION MARK, "\"") if the value contains more than one URI.
+
+#### <a href="#extension-rules" id="extension-rules" class="headerlink"></a> Rules for Extensions
+
+An extension **MAY** impose additional processing rules or further restrictions
+and it **MAY** define new object members as described below.
+
+An extension **MUST NOT** lessen or remove any processing rules, restrictions or
+object member requirements defined in this specification or other extensions.
+
+An extension **MAY** define new members within the document structure defined by
+this specification. The rules for extension members are covered
+[below](#extension-members).
+
+An extension **MAY** define new query parameters. The rules for extension-defined
+query parameters are covered [below](#extension-query-parameters).
+
+If an extension defines new query parameters or document members, the extension
+**MUST** define a namespace to guarantee that extensions will never conflict
+with the current or future versions of this specification. A namespace **MUST**
+meet all of the following conditions:
+
+- A namespace **MUST** contain at least one character.
+- A namespace **MUST** contain only these characters:
+  - U+0061 to U+007A, "a-z"
+  - U+0041 to U+005A, "A-Z"
+  - U+0030 to U+0039, "0-9"
+
+An extension **MUST NOT** define more than one namespace. The namespace used for
+all query parameters and documents members **MUST** be the same for any given
+extension.
+
+In the following example, an extension with the namespace `version` has
+specified a resource object member `version:id` to support per-resource
+versioning. This member might appear as follows:
+
+```json
+// ...
+{
+  "type": "articles",
+  "id": "1",
+  "version:id": "42",
+  "attributes": {
+    "title": "Rails is Omakase"
+  }
+}
+// ...
+```
+
+#### <a href="#profile-rules" id="profile-rules" class="headerlink"></a> Rules for Profiles
+
+The rules for profile usage are dictated by [RFC
+6906](https://tools.ietf.org/html/rfc6906).
+
+A profile **MAY** define document members, query parameters, and processing
+rules that are reserved for implementors.
+
+A server implementation that supports profile-defined query parameters **MUST
+NOT** support more than one profile per unique URL (including the entire query
+string).
```

Comment:
I forget exactly what our reasoning was for this statement although I'm sure we saw value in it. Was the intent for this statement to prevent two profiles from adding conflicting processing rules for the `filter` parameter?

IIUC, this goes too far and would preclude me from supporting this URL:

```
/users?foo=value&bar=value
```

where `foo` and `bar` are defined by two different profiles.

IOW, I _think_ this prevents an implementer from supporting >1 query parameter-defining profile at a time, which I don't believe we intended to do.

I tried and failed to reword this in a straightforward way. I feel like we've gone back and forth on this 1000 times, but perhaps we need a normative statement which says that profiles are not allowed to define query parameters of their own although they _are_ allowed to define processings rules for specification and extension defined query parameters whose interpretation is left to the implementor, like so:

```suggestion
A profile **MUST NOT** define any query parameters.

A profile **MUST NOT** alter or remove processing rules that have been defined
by this specification or by an [extension][extensions]. However, a profile
**MAY** define processing rules for query parameters whose processing rules
have been reserved for implementors to define at their discretion.

For example, a profile could define rules for interpreting [the `filter` query
parameter](#fetching-filtering), but it could not specify that relationship
names in [the `include` query parameter](#fetching-includes) are
space-separated instead of dot-separated.
```

I know that technically implementation-defined query parameters are part of implementation semantics, but I suspect that if we were to create a backwards-incompatible version 2 of JSON:API, we would leave extensions as the only means to define shareable query parameters.

Reply:
I think this refinement adds a lot of clarity to what's allowed for profiles vs. extensions with respect to query params. I struggle to think of a case of a new profile-defined query param that would not be better defined via an extension.

I'm going to accept your changes and also drop "query parameters" from:

> A profile **MAY** define document members, query parameters, and processing
> rules that are reserved for implementors.